lib/gen_stage.ex

```diff
@@ -1124,43 +1114,35 @@ defmodule GenStage do
   end
 
   @doc """
-  Asks the producer to send a notification to all consumers synchronously.
-
-  This call is synchronous and will return after the producer has either
-  sent the notification to all consumers or placed it in a buffer. In
-  other words, it guarantees the producer has handled the message but not
-  that the consumers have received it.
-
-  The given message will be delivered in the format
-  `{{producer_pid, subscription_tag}, message}`, where `message` is the message
-  given below. `subscription_tag` is the subscription tag passed to the
-  consumer's `c:handle_subscribe/4` callback in the `to_or_from` tuple.
-  Consumers that wish to verify the `subscription_tag` in their
-  `c:handle_info/2` implementation should store such subscription tag in
-  their state.
-
-  This function will return `:ok` as long as the notification request is
-  sent. It may return `{:error, :not_a_producer}` in case the stage is not
-  a producer.
+  Queues an info message that is delivered after all currently buffered events.
+
+  This call is synchronous and will return after the stage has queued
+  the info message. The message will be eventually handled by the
+  `handle_info/2` callback.
+
+  If the stage is a consumer, it does not have buffered events, so the
+  messaged is queued immediatelly.
+
+  This function will return `:ok` if the info message is successfully queued.
   """
-  @spec sync_notify(stage, term, timeout) :: :ok | {:error, :not_a_producer}
-  def sync_notify(stage, message, timeout \\ 5_000) do
-    call(stage, {:"$notify", message}, timeout)
+  @spec sync_info(stage, msg :: term(), timeout) :: :ok
+  def sync_info(stage, msg, timeout \\ 5_000) do
```

Comment:
For consistency, we should use `message` here since I changed a bunch of `msg` last time :bowtie: